app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/PoolContributionProcessor.kt

```diff
@@ -21,37 +21,40 @@ import rdx.works.profile.domain.accountsOnCurrentNetwork
 import javax.inject.Inject
 
 class PoolContributionProcessor @Inject constructor(
-    private val getResourcesUseCase: GetResourcesUseCase,
+    private val resolveAssetsFromAddressUseCase: ResolveAssetsFromAddressUseCase,
     private val getPoolDetailsUseCase: GetPoolDetailsUseCase,
     private val getProfileUseCase: GetProfileUseCase
 ) : PreviewTypeProcessor<DetailedManifestClass.PoolContribution> {
     override suspend fun process(summary: ExecutionSummary, classification: DetailedManifestClass.PoolContribution): PreviewType {
-        val resources = getResourcesUseCase(addresses = summary.involvedResourceAddresses).getOrThrow()
+        val assets = resolveAssetsFromAddressUseCase(
+            fungibleAddresses = summary.involvedFungibleAddresses,
+            nonFungibleIds = summary.involvedNonFungibleIds
+        ).getOrThrow()
         val involvedPools = getPoolDetailsUseCase(classification.poolAddresses.map { it.addressString() }.toSet()).getOrThrow()
-        val defaultDepositGuarantees = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
+        val defaultDepositGuarantee = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
         val accountsWithdrawnFrom = summary.accountWithdraws.keys
         val ownedAccountsWithdrawnFrom = getProfileUseCase.accountsOnCurrentNetwork().filter {
             accountsWithdrawnFrom.contains(it.address)
         }
-        val from = summary.extractWithdraws(ownedAccountsWithdrawnFrom, resources)
+        val from = summary.extractWithdraws(ownedAccountsWithdrawnFrom, assets.map { it.resource })
```

Comment:
I am a little confused. You can either extract withdraws from assets or from resources. If you choose resources, as you do here, you loose the ability to render for example fungibles as LSUs. It may not happen for a manifest to have to display LSUs in a pool contribution classification, but I am also not sure about all the states a summary can get. Have you understood the difference?

Can you explain why the change here?